ukvm/ukvm.c

```diff
@@ -577,93 +575,6 @@ void ukvm_port_nanosleep(uint8_t *mem, void *data, struct kvm_run *run)
     }
 }
 
-
-void ukvm_port_blkinfo(uint8_t *mem, void *data)
-{
-    uint32_t mem_off = *(uint32_t *) data;
-    struct ukvm_blkinfo *info = (struct ukvm_blkinfo *) (mem + mem_off);
-
-    info->sector_size = blkinfo.sector_size;
-    info->num_sectors = blkinfo.num_sectors;
-    info->rw = blkinfo.rw;
-}
-
-
-void ukvm_port_blkwrite(uint8_t *mem, void *data, int diskfd)
-{
-    uint32_t mem_off = *(uint32_t *) data;
-    struct ukvm_blkwrite *wr = (struct ukvm_blkwrite *) (mem + mem_off);
-    int ret;
-
-    wr->ret = -1;
-    if (wr->sector < blkinfo.num_sectors) {
-        lseek(diskfd, blkinfo.sector_size * wr->sector, SEEK_SET);
-        ret = write(diskfd, mem + (uint64_t) wr->data, wr->len);
-        assert(ret == wr->len);
-        wr->ret = 0;
-    }
-}
-
-
-void ukvm_port_blkread(uint8_t *mem, void *data, int diskfd)
-{
-    uint32_t mem_off = *(uint32_t *) data;
-    struct ukvm_blkread *rd = (struct ukvm_blkread *) (mem + mem_off);
-    int ret;
-
-    rd->ret = -1;
-    if (rd->sector < blkinfo.num_sectors) {
-        lseek(diskfd, blkinfo.sector_size * rd->sector, SEEK_SET);
-        ret = read(diskfd, mem + (uint64_t) rd->data, rd->len);
-        assert(ret == rd->len);
-        rd->ret = 0;
-    }
-}
-
-
-void ukvm_port_netinfo(uint8_t *mem, void *data)
-{
-    uint32_t mem_off = *(uint32_t *) data;
-    struct ukvm_netinfo *info = (struct ukvm_netinfo *) (mem + mem_off);
-
-    memcpy(info->mac_str, netinfo.mac_str, sizeof(netinfo.mac_str));
-}
-
-void ukvm_port_netwrite(uint8_t *mem, void *data, int netfd)
-{
-    uint32_t mem_off = *(uint32_t *) data;
-    struct ukvm_netwrite *wr = (struct ukvm_netwrite *) (mem + mem_off);
-    int ret;
-
-    wr->ret = 0;
-    ret = write(netfd, mem + (uint64_t) wr->data, wr->len);
-    assert(wr->len == ret);
-}
-
-
-void ukvm_port_netread(uint8_t *mem, void *data, int netfd)
-{
-    uint32_t mem_off = *(uint32_t *) data;
-    struct ukvm_netread *rd = (struct ukvm_netread *) (mem + mem_off);
-    struct timeval zero;
-    fd_set netset;
-    int ret;
-
-    FD_ZERO(&netset);
-    FD_SET(netfd, &netset);
-    zero.tv_sec = 0;
-    zero.tv_usec = 0;
-    ret = select(netfd + 1, &netset, NULL, NULL, &zero);
-    if (ret <= 0) {
-        rd->ret = -1;
-        return;
-    }
-
-    rd->len = read(netfd, mem + (uint64_t) rd->data, rd->len);
```

Comment:
`assert(rd->len > 0)` so that we don't drop errors on the floor? We should also check and handle `EINTR` on all these syscalls.